micro-rdk/src/common/robot.rs

```diff
@@ -183,10 +185,10 @@ impl LocalRobot {
             let cfg = &mut components[iter.next().unwrap()];
             if let Some(cfg) = cfg.as_ref() {
                 // capture the error and make it available to LocalRobot so it can be pushed in the logs?
-                if self
-                    .build_resource(cfg, board.clone(), board_key.clone(), &mut registry)
-                    .is_err()
+                if let Err(e) =
+                    self.build_resource(cfg, board.clone(), board_key.clone(), &mut registry)
                 {
+                    log::error!("{:?}", e);
```

Comment:
Neither the `RUST_LOG=debug` errors nor app.viam logs were helpful when troubleshooting init failure for camera. Not directly related to the ticket but may be useful to add either here or separate PR

Reply:
format the error reporting better in this case so there is a bit more context